src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/LanguageServiceHost.cs

```diff
@@ -165,6 +169,10 @@ private Task OnProjectFactoryCompletedAsync()
                     previousContextToDispose = _currentAggregateProjectContext;
                 }
 
+                // Force refresh the CPS active project configuration (needs UI thread).
+                await _commonServices.ThreadingService.SwitchToUIThread();
+                await _activeProjectConfigurationRefreshService.RefreshActiveProjectConfigurationAsync().ConfigureAwait(false);
+
                 // Create new project context.
```

Comment:
It doesn't but CreateProjectContextAsync needs to be invoked after the active configuration has been refreshed. Doesn't .ConfigureAwait(false) above indicate the scheduler is free to execute the remainder on a different context?